floater/generators.py

```diff
@@ -76,6 +81,16 @@ def __init__(self, xlim=None, ylim=None, dx=1., dy=1., model_grid=None, load_pat
             self.Ny = int(self.Ly / self.dy)
             self.x = self.xlim[0] + self.dx * np.arange(self.Nx) + self.dx/2
             self.y = self.ylim[0] + self.dy * np.arange(self.Ny) + self.dy/2
+            try:
+                zvect[0]
+            except (NameError,TypeError,AttributeError,KeyError) as e:
```

Comment:
I think there is a probably a cleaner way to do this. Rather than try to index the array and catching an error of this fails, what about
```python
if zvect not None:
    # do something
else:
    # do something else
```